assets/js/components/settings/SettingsRenderer.js

```diff
@@ -25,18 +25,21 @@ import { useEffect } from '@wordpress/element';
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
-import SettingsEdit from './SettingsEdit';
-import SettingsView from './SettingsView';
-import { STORE_NAME } from '../../datastore/constants';
+import { STORE_NAME as CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
 const { useSelect, useDispatch } = Data;
+const NullComponent = () => null;
+
+export default function SettingsRenderer( { slug, isOpen, isEditing } ) {
+	const storeName = `modules/${ slug }`;
+	const isDoingSubmitChanges = useSelect( ( select ) => select( storeName )?.isDoingSubmitChanges?.() );
+	const haveSettingsChanged = useSelect( ( select ) => select( storeName )?.haveSettingsChanged?.() );
+	const SettingsEdit = useSelect( ( select ) => select( CORE_MODULES ).getModule( slug )?.settingsEditComponent ) || NullComponent;
+	const SettingsView = useSelect( ( select ) => select( CORE_MODULES ).getModule( slug )?.settingsViewComponent ) || NullComponent;
```

Comment:
> Does `?.` work on `null` values like it does on `undefined`? Otherwise we'd run into problems here.

Yes; from the docs: 
![image](https://user-images.githubusercontent.com/1621608/97282036-ded1c980-1846-11eb-948c-f362a0cf459e.png)

> I think this would be easier to read by assigning only the raw selector return value into a `const module`, and then we could take care of the `settingsEditComponent` and `settingsViewComponent` from there.

I'll update it to use a single select instead 👍